# Aula 1: Introdução a variáveis usando Python

## As variáveis

A primeira coisa que temos que entender são as variáveis. Variáveis são entidades que o computador usa para armazenar informação. Elas tem três atributos principais: Um nome, um tipo e um valor. Veja: 




![teste](https://raw.githubusercontent.com/jp-guimaraes/intro_python/master/assets/var.png)



Por exemplo, podemos fazer:

In [1]:
x = 47;


Quando damos essa instrução ao computador estamos indiretamente dizendo: "Computador, reserve um espaço na memória que eu quero chamar de 'x'. Vou usar ele para guardar valores numéricos. A princípio ele vai guardar o número 47." ![teste](https://raw.githubusercontent.com/jp-guimaraes/intro_python/master/assets/var_x.png)

### Exibindo o conteúdo de uma variável armazenada na memória
Para exibir o conteúdo da variável 'x' na tela basta escrever x numa célula e executá-la. O atalho de teclado é shift + enter

In [2]:
x

47

Sempre que eu me referir a 'x', o computador vai na sua memória, no endereço que separou para o 'x', e lê esse resultado.
Mas, como o próprio nome diz, trata-se de um valor variável. Para mudar o valor que está armazenado nessa memória basta fazer uma nova operação de atribuição. Veja:

In [3]:
x = 2


Agora a posição de memória referenciada pelo nome 'x' guarda o número 2. 

Em python, as variáveis do tipo numéricas podem ser tanto inteiras quanto fracionárias. Ao contrário de outras linguagens, não há uma diferenciação na declaração dela. Então, se quisermos que a variável 'x' passe a representar o preço do Kg de feijão, que custa em torno de R$ 11.20 (onze reais e vinte centavos), por exemplo.

In [4]:
x = 11.20


Imprimindo o conteúdo da variável x, obtemos:

In [5]:
x

11.2

## Operando variáveis 

Você pode notar que para inicializar ou valores a uma variável numérica, o símbolo '=' foi utilizado. Ele é chamado de operador de atribuição. Existem outros tipos de operadores. Para variáveis numéricas destacam-se os operadores aritméticos. Veja alguns exemplos com os operadores soma (+), subtração(-), multiplicação (*) e divisão (/):


In [6]:
x+1

12.2

In [7]:
x-13.4

-2.200000000000001

In [8]:
x*3.14

35.168

In [9]:
x/(8)

1.4

## Outro tipo de variável: As strings
Outro tipo de variável muito comum são as strings. Stings são cadeias de caracteres utilizadas para armazenar palavras. Em Python, para atribuir uma string a uma variável basta usar as áspas. Exemplo:


In [10]:
x = "bola"


Como pode-se notar, não há nenhuma diferenciação da atribuição numérica a strings em python. 'x' agora é a string 'bola'! 

In [11]:
x

'bola'

A forma de leitura é a mesma: "Computador, reserve um espaço de memória que eu quero chamar de 'x' e guarde lá dentro a palavra 'bola' ". A informação anterior guardada na memória, no caso o valor 11.2 é descartada.

## Variáveis Booleanas
As variáveis booleanas também estão presentes em Python. Em geral, modela-se informações do mundo real que expressam antagonismo por uma variável binária no computador, por exemplo: cheio e vazio, verdadeiro e falso, 5V e 0V, pertence ou não pertence, contém e não contém, etc. Em python usa-se o clássico "Verdadeiro e Falso" mas em inglês, ou seja, True e False. Assim, é possível fazer as seguintes atribuições: 

In [12]:
x = True

ou então


In [13]:
y = False

### Operando variáveis booleanas

As variáveis booleanas podem ser submetidas a operadores lógicos. Veja:

In [14]:
x and y

False

In [15]:
x or y

True

In [16]:
not x

False

## Fazendo perguntas

É possível usar operadores lógicos em variável numéricas para obter como resultado variáveis booleanas. Para isso são usados operados tais como > (maior) , < (menor), maior ou igual (>=), menor igual (<=), igual (==) e diferente (!=). Veja alguns exemplos:

In [17]:
3<4

True

In [18]:
4<3

False

In [19]:
4<=4

True

In [20]:
2 == 2

True

In [21]:
3 != 3

False

In [22]:
4> 3.9999999999

True

Faça alguns testes mudando os valores nas perguntas acima, observando sempre os resultados.